From e15962cc0cf487a85c4c30a2df0c4fd6dabe66de Mon Sep 17 00:00:00 2001
From: Pavel Tikhomirov <ptikhomirov@virtuozzo.com>
Date: Fri, 29 Oct 2021 10:35:28 +0300
Subject: [PATCH 039/249] clang-format: make x86_ins_capability_mask
 human-readable

There is no option in clang not to merge as much binary operands as it
fits in column limit, but here we need each bit on new line to make it
readable, so let's disable clang-format for x86_ins_capability_masks.

Signed-off-by: Pavel Tikhomirov <ptikhomirov@virtuozzo.com>
---
 criu/arch/x86/cpu.c | 139 ++++++++++++++++++++++++++++----------------
 1 file changed, 89 insertions(+), 50 deletions(-)

diff --git a/criu/arch/x86/cpu.c b/criu/arch/x86/cpu.c
index d02f4abd5..b3a7ca636 100644
--- a/criu/arch/x86/cpu.c
+++ b/criu/arch/x86/cpu.c
@@ -107,64 +107,103 @@ int cpu_dump_cpuinfo(void)
 
 #define __ins_bit(__l, __v) (1u << ((__v)-32u * (__l)))
 
+// clang-format off
 static uint32_t x86_ins_capability_mask[NCAPINTS] = {
-	[CPUID_1_EDX] = __ins_bit(CPUID_1_EDX, X86_FEATURE_FPU) | __ins_bit(CPUID_1_EDX, X86_FEATURE_TSC) |
-			__ins_bit(CPUID_1_EDX, X86_FEATURE_CX8) | __ins_bit(CPUID_1_EDX, X86_FEATURE_SEP) |
-			__ins_bit(CPUID_1_EDX, X86_FEATURE_CMOV) | __ins_bit(CPUID_1_EDX, X86_FEATURE_CLFLUSH) |
-			__ins_bit(CPUID_1_EDX, X86_FEATURE_MMX) | __ins_bit(CPUID_1_EDX, X86_FEATURE_FXSR) |
-			__ins_bit(CPUID_1_EDX, X86_FEATURE_XMM) | __ins_bit(CPUID_1_EDX, X86_FEATURE_XMM2),
-
-	[CPUID_8000_0001_EDX] = __ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_SYSCALL) |
-				__ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_MMXEXT) |
-				__ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_RDTSCP) |
-				__ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_3DNOWEXT) |
-				__ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_3DNOW),
-
-	[CPUID_LNX_1] = __ins_bit(CPUID_LNX_1, X86_FEATURE_REP_GOOD) | __ins_bit(CPUID_LNX_1, X86_FEATURE_NOPL),
-
-	[CPUID_1_ECX] = __ins_bit(CPUID_1_ECX, X86_FEATURE_XMM3) | __ins_bit(CPUID_1_ECX, X86_FEATURE_PCLMULQDQ) |
-			__ins_bit(CPUID_1_ECX, X86_FEATURE_MWAIT) | __ins_bit(CPUID_1_ECX, X86_FEATURE_SSSE3) |
-			__ins_bit(CPUID_1_ECX, X86_FEATURE_CX16) | __ins_bit(CPUID_1_ECX, X86_FEATURE_XMM4_1) |
-			__ins_bit(CPUID_1_ECX, X86_FEATURE_XMM4_2) | __ins_bit(CPUID_1_ECX, X86_FEATURE_MOVBE) |
-			__ins_bit(CPUID_1_ECX, X86_FEATURE_POPCNT) | __ins_bit(CPUID_1_ECX, X86_FEATURE_AES) |
-			__ins_bit(CPUID_1_ECX, X86_FEATURE_XSAVE) | __ins_bit(CPUID_1_ECX, X86_FEATURE_OSXSAVE) |
-			__ins_bit(CPUID_1_ECX, X86_FEATURE_AVX) | __ins_bit(CPUID_1_ECX, X86_FEATURE_F16C) |
-			__ins_bit(CPUID_1_ECX, X86_FEATURE_RDRAND),
+	[CPUID_1_EDX] =
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_FPU) |
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_TSC) |
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_CX8) |
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_SEP) |
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_CMOV) |
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_CLFLUSH) |
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_MMX) |
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_FXSR) |
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_XMM) |
+	__ins_bit(CPUID_1_EDX, X86_FEATURE_XMM2),
+
+	[CPUID_8000_0001_EDX] =
+	__ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_SYSCALL) |
+	__ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_MMXEXT) |
+	__ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_RDTSCP) |
+	__ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_3DNOWEXT) |
+	__ins_bit(CPUID_8000_0001_EDX, X86_FEATURE_3DNOW),
+
+	[CPUID_LNX_1] =
+	__ins_bit(CPUID_LNX_1, X86_FEATURE_REP_GOOD) |
+	__ins_bit(CPUID_LNX_1, X86_FEATURE_NOPL),
+
+	[CPUID_1_ECX] =
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_XMM3) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_PCLMULQDQ) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_MWAIT) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_SSSE3) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_CX16) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_XMM4_1) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_XMM4_2) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_MOVBE) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_POPCNT) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_AES) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_XSAVE) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_OSXSAVE) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_AVX) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_F16C) |
+	__ins_bit(CPUID_1_ECX, X86_FEATURE_RDRAND),
 
 	[CPUID_8000_0001_ECX] =
-		__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_ABM) | __ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_SSE4A) |
-		__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_MISALIGNSSE) |
-		__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_3DNOWPREFETCH) |
-		__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_XOP) | __ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_FMA4) |
-		__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_TBM),
+	__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_ABM) |
+	__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_SSE4A) |
+	__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_MISALIGNSSE) |
+	__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_3DNOWPREFETCH) |
+	__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_XOP) |
+	__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_FMA4) |
+	__ins_bit(CPUID_8000_0001_ECX, X86_FEATURE_TBM),
 
 	[CPUID_7_0_EBX] =
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_FSGSBASE) | __ins_bit(CPUID_7_0_EBX, X86_FEATURE_BMI1) |
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_HLE) | __ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX2) |
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_BMI2) | __ins_bit(CPUID_7_0_EBX, X86_FEATURE_ERMS) |
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_RTM) | __ins_bit(CPUID_7_0_EBX, X86_FEATURE_MPX) |
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512F) | __ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512DQ) |
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_RDSEED) | __ins_bit(CPUID_7_0_EBX, X86_FEATURE_ADX) |
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_CLFLUSHOPT) | __ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512PF) |
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512ER) | __ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512CD) |
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_SHA_NI) | __ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512BW) |
-		__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512VL),
-
-	[CPUID_D_1_EAX] = __ins_bit(CPUID_D_1_EAX, X86_FEATURE_XSAVEOPT) |
-			  __ins_bit(CPUID_D_1_EAX, X86_FEATURE_XSAVEC) | __ins_bit(CPUID_D_1_EAX, X86_FEATURE_XGETBV1),
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_FSGSBASE) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_BMI1) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_HLE) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX2) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_BMI2) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_ERMS) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_RTM) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_MPX) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512F) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512DQ) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_RDSEED) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_ADX) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_CLFLUSHOPT) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512PF) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512ER) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512CD) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_SHA_NI) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512BW) |
+	__ins_bit(CPUID_7_0_EBX, X86_FEATURE_AVX512VL),
+
+	[CPUID_D_1_EAX] =
+	__ins_bit(CPUID_D_1_EAX, X86_FEATURE_XSAVEOPT) |
+	__ins_bit(CPUID_D_1_EAX, X86_FEATURE_XSAVEC) |
+	__ins_bit(CPUID_D_1_EAX, X86_FEATURE_XGETBV1),
 
 	[CPUID_7_0_ECX] =
-		__ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512VBMI) | __ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512_VBMI2) |
-		__ins_bit(CPUID_7_0_ECX, X86_FEATURE_GFNI) | __ins_bit(CPUID_7_0_ECX, X86_FEATURE_VAES) |
-		__ins_bit(CPUID_7_0_ECX, X86_FEATURE_VPCLMULQDQ) | __ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512_VNNI) |
-		__ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512_BITALG) | __ins_bit(CPUID_7_0_ECX, X86_FEATURE_TME) |
-		__ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512_VPOPCNTDQ) | __ins_bit(CPUID_7_0_ECX, X86_FEATURE_RDPID),
-
-	[CPUID_8000_0008_EBX] = __ins_bit(CPUID_8000_0008_EBX, X86_FEATURE_CLZERO),
-
-	[CPUID_7_0_EDX] = __ins_bit(CPUID_7_0_EDX, X86_FEATURE_AVX512_4VNNIW) |
-			  __ins_bit(CPUID_7_0_EDX, X86_FEATURE_AVX512_4FMAPS),
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512VBMI) |
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512_VBMI2) |
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_GFNI) |
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_VAES) |
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_VPCLMULQDQ) |
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512_VNNI) |
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512_BITALG) |
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_TME) |
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_AVX512_VPOPCNTDQ) |
+	__ins_bit(CPUID_7_0_ECX, X86_FEATURE_RDPID),
+
+	[CPUID_8000_0008_EBX] =
+	__ins_bit(CPUID_8000_0008_EBX, X86_FEATURE_CLZERO),
+
+	[CPUID_7_0_EDX] =
+	__ins_bit(CPUID_7_0_EDX, X86_FEATURE_AVX512_4VNNIW) |
+	__ins_bit(CPUID_7_0_EDX, X86_FEATURE_AVX512_4FMAPS),
 };
+// clang-format on
 
 #undef __ins_bit
 
-- 
2.35.1

